### PR TITLE
ci: change Travis to GitHub actions

### PR DESCRIPTION
**Pull request by Divlo**
**What changes did you make? (Give an overview)**

- [x] Change travis configuration to GitHub Actions for continuous integration

This allows us to be consistent across the organization, we're slowly migrating to GitHub Actions.

Related: https://github.com/standard/eslint-config-standard-with-typescript/pull/530, https://github.com/standard/standard-engine/pull/233, https://github.com/standard/standard-engine/pull/233

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        with:
+          node-version: 'lts/*'
+
+      - name: 'Cache Node dependencies'
+        uses: 'actions/cache@v2.1.6'
+        with:
+          path: '~/.npm'
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
```
Comment (voxpelli) [inline .github/workflows/ci.yml:26]:
```suggestion
      - name: 'Use Node.js'
        uses: 'actions/setup-node@v2.4.0'
        with:
          node-version: 'lts/*'
          cache: 'npm'
```

Nowadays its enough[ to do `cache: 'npm'`](https://github.com/actions/setup-node/releases/tag/v2.2.0) 🎉

**Review by voxpelli — COMMENTED** [review]
We can simplify it a bit nowadays 🙂 

I also wonder if we maybe would be helped by syncing these scripts using `ghat`? I do that for my repos, using my ghat templates here: https://github.com/voxpelli/ghatemplates

That way one don't have to remember to update the action, one can simply run a sync every now and then and see if some changes pop up.

And it's easier to know which of the action definitions that are the most modern and the one to copy from 🙂

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Divlo) [reply .github/workflows/ci.yml:26]: That's not working without `package-lock.json` badly.

Comment (voxpelli) [reply .github/workflows/ci.yml:26]: Oh, had totally missed that part, never mind me then

**Review by voxpelli — APPROVED** [review]

**Comment by Divlo** [comment]
@voxpelli 

> I also wonder if we maybe would be helped by syncing these scripts using ghat? I do that for my repos, using my ghat templates here: voxpelli/ghatemplates

I didn't know the existence of [ghat](https://github.com/fregante/ghat), great suggestion!
I think you can open a new meta issue on [standard/standard](https://github.com/standard/standard], so we could create a new repo to maintain GitHub Actions for the whole organization.

Meanwhile, that's probably out of the scope of this PR. :smile:

**Pull request merged**